derive/src/trait_bounds.rs

```diff
@@ -28,29 +28,34 @@ use syn::{
 /// associated types (e.g. `T::A: TypeInfo`), correctly dealing with
 /// self-referential types.
 pub fn add(input_ident: &Ident, generics: &mut Generics, data: &syn::Data) -> Result<()> {
-    let ty_params = generics.type_params_mut().fold(Vec::new(), |mut acc, p| {
-        p.bounds.push(parse_quote!(::scale_info::TypeInfo));
-        p.bounds.push(parse_quote!('static));
-        acc.push(p.ident.clone());
-        acc
-    });
+    let ty_params_ids = generics
+        .type_params()
+        .map(|type_param| type_param.ident.clone())
+        .collect::<Vec<Ident>>();
 
-    if ty_params.is_empty() {
+    if ty_params_ids.is_empty() {
         return Ok(())
     }
 
-    let types = collect_types_to_bind(input_ident, data, &ty_params)?;
-
-    if !types.is_empty() {
-        let where_clause = generics.make_where_clause();
+    let types = collect_types_to_bind(input_ident, data, &ty_params_ids)?;
+    let generics_clone = generics.clone();
```

Comment:
This is a bit nasty; what's a better way?

Reply:
Could do something like `let type_params = generics.type_params().cloned().collect::<Vec<_>>();` and then use `type_params.into_iter().for_each(...` in line 50, could also remove the clones in the closure